scripts/git-branch.ps1

```diff
@@ -0,0 +1,135 @@
+
+# create a new release branch with annoated tag and push to git origin
+
+[CmdletBinding()]
+param
+(
+    [Parameter(Mandatory = $false)]
+    [string]$Name,
+    
+    [Parameter(Mandatory = $false)]
+    [string]$Email
+)
+
+function TagExists($TagName, [switch]$Remote) {
+    
+    # look for the name in local refs/tags
+
+    if ($Remote -eq $true) {
+
+        foreach ($tag in $(git ls-remote --tags)) {
+
+            $name = Split-Path $tag -Leaf
+    
+            if ($name -eq $TagName) {
+                return $true
+            }
+        }
+    }
+    else {
+
+        foreach ($tag in $(git tag)) {
+
+            if ($tag -eq $TagName) {
+                return $true
+            }
+        }
+    }
+
+    return $false
+}
+
+function ExitOrContinue {
+
+    # stop only if prior git command failed
+
+    Write-Host "Last exit code is ($LASTEXITCODE)"
+
+    if ($LASTEXITCODE -ne 0) {
+        exit $LASTEXITCODE  # terminate
+    }
+}
+
+function CurrentBranch() {
+
+    # find current branch name from git output
+
+    foreach ($line in $(git branch)) {
+
+        # git branch output marks the current branch with a leading astrisk
+
+        if ($line.StartsWith('*')) {
+            return $line
+        }
+    }
+
+    throw "Current branch not found"
+}
+
+
+
+
+# start:
+
+$date = Get-Date -UFormat "%Y-%m-%d"
+$branch = "release_$date"
+$sources = $env:BUILD_SOURCESDIRECTORY
+
+Write-Output "Set current directory to '$sources'"
+Set-Location $sources
+
+git config user.email $Email
+git config user.name $Name
+
+$message = "Bump package versions for release $date"
+
+Write-Output "Create new branch '$($branch)'"
+
+git checkout -b $branch
+ExitOrContinue  # exit if we can't create the branch
+
+$lastCommitId = git rev-list --tags --max-count=1
+$lastTag = git describe $lastCommitId  # By default git describe only shows annotated tags.
+
+Write-Output "Print commits from $lastTag..$branch"
+
+git --no-pager log $lastTag..$branch --name-status
+
+Write-Output "Add changes to commit"
+
+git add .
+ExitOrContinue  # exit if we can't add the changes
+
+git commit -m $message
+ExitOrContinue  # exit if we can't commit the changes
+
+if ($(CurrentBranch) -eq '* master') {
```

Comment:
>*  [](start = 26, length = 2)

to echo the previous comment i think that's a little weird. I can live with it but if we can bring back the code that removed the * and trims the spaces i think it'll look better.